### ISSUE TITLE
cl : Command line warning D9025 : overriding '/W3' with '/W4'

### ISSUE THREAD

**Issue by jhasse** [kcat/openal-soft#484]
When configuring OpenAL Soft on Windows with `cmake -Bbuild -GNinja` I'm getting the following warning on every compiler invocation:

```
cl : Command line warning D9025 : overriding '/W3' with '/W4'
```

It seems that CMake adds `/W3` implicitly (only when using the Ninja generator?). I think the best way to fix this would be to add the compile options with a CMake function which checks for existing `/W[0-4]` flags and replace them - it requires quite a few changes to the CMakeLists.txt though.

**Comment by kcat** [comment]
Is it possible to find out which CMake variable(s) are trying to add `/W3`? https://github.com/kcat/openal-soft/blob/master/CMakeLists.txt#L223 swaps out `/MD` for `/MT` when static linking the VCRT, so perhaps something similar would work for `/W3`.

**Comment by jhasse** [comment]
I've checked and it's `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS`.

**Comment by kcat** [comment]
Commit 36ecea458a5625df0e07dc4f6fb1c1280d348b3a tries to get rid of the extraneous `/W3`. Does that work there?

**Comment by jhasse** [comment]
I guess `-DNDEBUG` is a typo? If I change that to `/W3` it works :)

**Comment by kcat** [comment]
Indeed, fixed in commit 9e7c81649869432f99f8fd4780af1dbda007a219. Thanks.

**Comment by jhasse** [comment]
thank you!
